src/guide/advanced/permissions.md

```diff
@@ -0,0 +1,286 @@
+   `Grant` special instructions, and the received permissions could be
+   removed using `Revoke` instruction.
+
+## Permission Tokens
+
+The following permission tokens are pre-configured in Iroha 2:
+
+| Permission Token                                                                            | Category         | Operation        |
+| ------------------------------------------------------------------------------------------- | ---------------- | ---------------- |
+| [`CanSetKeyValueInUserMetadata`](#cansetkeyvalueinusermetadata)                             | Account          | Set key value    |
+| [`CanBurnUserAssets`](#canburnuserassets)                                                   | Asset            | Burn             |
+| [`CanSetKeyValueInUserAssets`](#cansetkeyvalueinuserassets)                                 | Asset            | Set key value    |
+| [`CanRemoveKeyValueInUserAssets`](#canremovekeyvalueinuserassets)                           | Asset            | Remove key value |
+| [`CanTransferUserAssets`](#cantransferuserassets)                                           | Asset            | Transfer         |
+| [`CanTransferOnlyFixedNumberOfTimesPerPeriod`](#cantransferonlyfixednumberoftimesperperiod) | Asset            | Transfer         |
+| [`CanMintUserAssetDefinitions`](#canmintuserassetdefinitions)                               | Asset Definition | Mint             |
+| [`CanBurnAssetWithDefinition`](#canburnassetwithdefinition)                                 | Asset Definition | Burn             |
+| [`CanUnregisterAssetWithDefinition`](#canunregisterassetwithdefinition)                     | Asset Definition | Unregister       |
+| [`CanSetKeyValueInAssetDefinition`](#cansetkeyvalueinassetdefinition)                       | Asset Definition | Set key value    |
+| [`CanRemoveKeyValueInAssetDefinition`](#canremovekeyvalueinassetdefinition)                 | Asset Definition | Remove key value |
+| [`CanRegisterDomains`](#canregisterdomains)                                                 | Domain           | Register         |
+
+### `CanMintUserAssetDefinitions`
+
+With `CanMintUserAssetDefinitions`, a user can register and mint assets
+with the corresponding asset definition.
+
+```rust
+let mut genesis = RawGenesisBlock::new(
+    "alice".parse(),
+    "wonderland".parse(),
+    get_key_pair().public_key().clone(),
+);
+let rose_definition_id =
+    <AssetDefinition as Identifiable>::Id::from_str("rose#wonderland");
+let alice_id =
+    <Account as Identifiable>::Id::from_str("alice@wonderland");
+
+// Create a new `CanMintUserAssetDefinitions` permission token
+// to mint rose assets (`rose_definition_id`)
+let mint_rose_permission: PermissionToken =
+    CanMintUserAssetDefinitions::new(rose_definition_id.clone()).into();
+
+// Grant Alice permission to mint rose assets
+genesis.transactions[0]
+    .isi
+    .push(GrantBox::new(mint_rose_permission, alice_id.clone()).into());
+```
```

Comment:
```suggestion
```rust
let mut genesis = RawGenesisBlock::new(
    "alice".parse(),
    "wonderland".parse(),
    get_key_pair().public_key().clone(),
);
let rose_definition_id =
    <AssetDefinition as Identifiable>::Id::from_str("rose#wonderland")?;
let alice_id =
    <Account as Identifiable>::Id::from_str("alice@wonderland")?;

// Create a new `CanMintUserAssetDefinitions` permission token
// to mint rose assets (`rose_definition_id`)
let mint_rose_permission: PermissionToken =
    CanMintUserAssetDefinitions::new(rose_definition_id).into();

// Grant Alice permission to mint rose assets
genesis.transactions[0]
    .isi
    .push(GrantBox::new(mint_rose_permission, alice_id).into());
```
```

Reply:
`from_str()` is a failable operation so we need to add i.e. `?` here to make *Rust* unwrap the value if the operation has succeed or cancel current function an forward an error if the operation has failed

Reply:
Also no need for extra-clonning `alice_id` and `rose_definition_id` as it not used anywhere anymore.

Note that this is not the case for `get_key_pair().public_key().clone()`. We need `clone()` here, because `public_key()` returns a reference to key, but `RawGenesisBlock::new()` expects the key it-self so we do clonning to retrieve it

Reply:
Please, check the other code in this file for the same things

Reply:
thanks for the explanation
by the way, I was taking these examples from the source code so there might be some changes needed there as well

Reply:
Which doc-test specifically have you taken this from?

Reply:
Ok I realised I've removed the `.expect("Valid")` ending, so it's not a complete copy-paste.

this particular example is from test network: https://github.com/hyperledger/iroha/blob/iroha2-dev/core/test_network/src/lib.rs#L92

but I've also took examples from `permissions_validators/src/public_blockchain/mod.rs`, e.g. about unregister instruction: https://github.com/hyperledger/iroha/blob/iroha2-dev/permissions_validators/src/public_blockchain/mod.rs#L291